nncf/quantization/algorithms/weight_compression/scale_estimation.py

```diff
@@ -0,0 +1,301 @@
+    """
+
+    def __init__(
+        self,
+        model: TModel,
+        name_to_node_mapping: Dict[str, Any],
+        all_weight_params: List[WeightCompressionParameters],
+        nodes_to_compress: List[NNCFNode],
+        activations: Optional[Dict[str, TTensor]] = None,
+        subset_size: int = 32,
+        initial_steps: int = 5,
+        scale_steps: int = 10,
+        weight_penalty: float = -1.0,
+    ):
+        """
+        :param model: Model for applying algorithm.
+        :param name_to_node_mapping: Name to node mapping for updating node weights.
+        :param all_weight_params: List of all weight parameters.
+        :param nodes_to_compress: List of nodes for processing.
+        :param activations: The input activations of the layers considered for compression.
+        :param subset_size: The number of samples for scale estimation.
+        :param initial_steps: The number of the steps for absmax scale rectification.
+        :param scale_steps: The number of the steps for grid search scale rectification
+                            from 1.0 to 1.0 - 0.05 * scale_step.
+        :param weight_penalty: coefficient for penalty between fp and compressed weights. If -1 then doesn't apply.
+        """
+        super().__init__()
+        self.name_to_node_mapping = name_to_node_mapping
+        self._all_weight_params = all_weight_params
+        self._nodes_to_compress = nodes_to_compress
+        self._activations = activations
+        self._subset_size = subset_size
+        self._initial_steps = initial_steps
+        self._scale_steps = scale_steps
+        self._weight_penalty = weight_penalty
+
+        self._set_backend_entity(model)
+
+    @property
+    def available_backends(self) -> List[BackendType]:
+        return [BackendType.OPENVINO]
+
+    def _set_backend_entity(self, model: TModel) -> None:
+        """
+        Creates a helper class with a backed-specific logic of the algorithm.
+
+        :param model: Backend-specific input model.
+        :param all_weight_params: List of all weight parameters.
+        :param nodes_to_compress: List of nodes for processing.
+        :param activations: The input activations of the layers considered for compression.
+        """
+
+        model_backend = get_backend(model)
+        if model_backend == BackendType.OPENVINO:
+            from nncf.quantization.algorithms.weight_compression.openvino_backend import OVWeightCompressionAlgoBackend
+
+            self._backend_entity = OVWeightCompressionAlgoBackend(model, self.name_to_node_mapping)
+        else:
+            raise RuntimeError(
+                "Cannot return backend-specific AWQ entity because {} is not supported!".format(model_backend.value)
+            )
+
+    def apply(
+        self,
+        model: TModel,
+        graph: NNCFGraph,
+        statistic_points: Optional[StatisticPointsContainer] = None,
+        dataset: Optional[Dataset] = None,
+    ) -> TModel:
+        """
```

Comment:
This function is 180 lines long and implements several critical steps at once (weights reshaping/preparing, rectification of initial scale, rectification of scale based on grid search). I believe stages separation in private functions with minimal description will not only improve readability, but allow to unit test separate parts independently

Reply:
@daniil-lyakhov  Are you expecting something like this https://github.com/andreyanufr/nncf/commit/2c9924807f126f0ab7b4e919513efa33d8678c12 ?

Reply:
I suggest to create an issue for that